sys/net/gnrc/link_layer/gnrc_lorawan/gnrc_lorawan.c

```diff
@@ -0,0 +1,3360 @@
+    {
+        netdev->driver->set(netdev, NETOPT_STATE, &state, sizeof(netopt_state_t));
+    }
+    else
+    {
+        on_rx2_timeout(loradev);
+    }
+}
+
+void on_rx_error(netdev2_lorawan_t *loradev)
+{
+    netopt_state_t state = NETOPT_STATE_SLEEP;
+    netdev2_t *netdev = (netdev2_t*) loradev;
+    if( loradev->LoRaMacDeviceClass != CLASS_C )
+    {
+        netdev->driver->set(netdev, NETOPT_STATE, &state, sizeof(netopt_state_t));
+    }
+    else
+    {
+        on_rx2_timeout(loradev);
+    }
+}
+
+void on_rx_timeout(netdev2_lorawan_t *loradev)
+{
+    netopt_state_t state = NETOPT_STATE_SLEEP;
+    netdev2_t *netdev = (netdev2_t*) loradev;
+    if( loradev->LoRaMacDeviceClass != CLASS_C )
+    {
+        netdev->driver->set(netdev, NETOPT_STATE, &state, sizeof(netopt_state_t));
+    }
+    else
+    {
+        on_rx2_timeout(loradev);
+    }
+
+    if( loradev->rx_slot == 1 )
+    {
+        if(loradev->join_req_sent && loradev->lorawan.tx_rx.nwk_status == false && loradev->ChannelsNbRepCounter >= loradev->LoRaMacParams.ChannelsNbRep)
+        {
+            retransmit_join_req(loradev);
+        }
+        _finish_rx(loradev);
+    }
+}
+
+void _finish_rx(netdev2_lorawan_t *loradev)
+{
+    loradev->ChannelsNbRepCounter = 0;
+
+    loradev->AdrAckCounter++;
+    if( loradev->IsUpLinkCounterFixed == false )
+    {
+        loradev->uplink_counter++;
+    }
+
+}
+
+void on_tx_delay_timeout(netdev2_lorawan_t *loradev)
+{
+    //TimerStop( &TxDelayedTimer );
+    xtimer_remove(&loradev->TxDelayedTimer.dev);
+
+    if(loradev->join_req_sent)
+    {
+        gnrc_lorawan_send_join_request(loradev);
+    }
+    else
+    {
+        _schedule_next_tx(loradev);
+    }
+}
+
+void on_rx1_timeout(netdev2_lorawan_t *loradev)
+{
+    uint16_t symbTimeout = 5; // DR_2, DR_1, DR_0
+    int8_t datarate = 0;
+    uint32_t bandwidth = 0; // LoRa 125 kHz
+    netdev2_t *netdev = (netdev2_t*) loradev;
+
+    //TimerStop( &RxWindowTimer1 );
+    xtimer_remove(&loradev->RxWindowTimer1.dev);
+    loradev->rx_slot = 0;
+
+    netopt_state_t state = NETOPT_STATE_STANDBY;
+    if( loradev->LoRaMacDeviceClass == CLASS_C )
+    {
+        netdev->driver->set(netdev, NETOPT_STATE, &state, sizeof(netopt_state_t));
+    }
+
+#if defined( USE_BAND_433 ) || defined( USE_BAND_780 ) || defined( USE_BAND_868 )
+    datarate = loradev->LoRaMacParams.ChannelsDatarate - loradev->LoRaMacParams.Rx1DrOffset;
+    if( datarate < 0 )
+    {
+        datarate = DR_0;
+    }
+
+    // For higher datarates, we increase the number of symbols generating a Rx Timeout
+    if( ( datarate == DR_3 ) || ( datarate == DR_4 ) )
+    { // DR_4, DR_3
+        symbTimeout = 8;
+    }
+    else if( datarate == DR_5 )
+    {
+        symbTimeout = 10;
+    }
+    else if( datarate == DR_6 )
+    {// LoRa 250 kHz
+        bandwidth  = 1;
+        symbTimeout = 14;
+    }
+    _rx_window_setup(loradev, channels[Channel].Frequency, datarate, bandwidth, symbTimeout, false );

```

Comment:
Compile error on that line in case we using USE_BAND_868. 
`error: 'Channel' undeclared (first use in this function)`

It's better to test your implementation for compilation errors for other than 915MHz bands.